bindings/aws/ses/ses.go

```diff
@@ -0,0 +1,188 @@
+)
+
+const (
+	// The character encoding for the email.
+	CharSet = "UTF-8"
+)
+
+// AWSSES is an AWS SNS binding
+type AWSSES struct {
+	metadata sesMetadata
+	logger   logger.Logger
+}
+
+type sesMetadata struct {
+	Region       string `json:"region"`
+	AccessKey    string `json:"accessKey"`
+	SecretKey    string `json:"secretKey"`
+	SessionToken string `json:"sessionToken"`
+	EmailFrom    string `json:"emailFrom"`
+	EmailTo      string `json:"emailTo"`
+	Subject      string `json:"subject"`
+	EmailCc      string `json:"emailCc"`
+	EmailBcc     string `json:"emailBcc"`
+}
+
+// NewAWSSES creates a new AWSSES binding instance
+func NewAWSSES(logger logger.Logger) *AWSSES {
+	return &AWSSES{logger: logger}
+}
+
+// Init does metadata parsing
+func (a *AWSSES) Init(metadata bindings.Metadata) error {
+	// Parse input metadata
+	meta, err := a.parseMetadata(metadata)
+	if err != nil {
+		return err
+	}
+
+	a.metadata = meta
+
+	return nil
+}
+
+func (a *AWSSES) Operations() []bindings.OperationKind {
+	return []bindings.OperationKind{bindings.CreateOperation}
+}
+
+func (a *AWSSES) parseMetadata(meta bindings.Metadata) (sesMetadata, error) {
+	sesMeta := sesMetadata{}
+
+	if meta.Properties["region"] == "" || meta.Properties["accessKey"] == "" ||
+		meta.Properties["secretKey"] == "" {
+		return sesMeta, errors.New("SES binding error: region, accessKey or secretKey fields are required in metadata")
+	}
+
+	sesMeta.Region = meta.Properties["region"]
+	sesMeta.AccessKey = meta.Properties["accessKey"]
+	sesMeta.SecretKey = meta.Properties["secretKey"]
+	sesMeta.SessionToken = meta.Properties["sessionToken"]
+
+	// Optional properties, these can be set on a per request basis
+	sesMeta.EmailTo = meta.Properties["emailTo"]
+	sesMeta.EmailFrom = meta.Properties["emailFrom"]
+	sesMeta.Subject = meta.Properties["subject"]
+	sesMeta.EmailCc = meta.Properties["emailCc"]
+	sesMeta.EmailBcc = meta.Properties["emailBcc"]
+
+	return sesMeta, nil
+}
+
+func (a *AWSSES) Invoke(req *bindings.InvokeRequest) (*bindings.InvokeResponse, error) {
+	metadata := a.metadata.mergeWithRequestMetadata(req)
+
+	if metadata.EmailFrom == "" {
+		return nil, fmt.Errorf("SES binding error: emailFrom property not supplied in configuration- or request-metadata")
+	}
+	if metadata.EmailTo == "" {
+		return nil, fmt.Errorf("SES binding error: emailTo property not supplied in configuration- or request-metadata")
+	}
+	if metadata.Subject == "" {
+		return nil, fmt.Errorf("SES binding error: subject property not supplied in configuration- or request-metadata")
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(metadata.Region),
+		Credentials: credentials.NewStaticCredentials(metadata.AccessKey, metadata.SecretKey, metadata.SessionToken),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("SES binding error: error creating AWS session %w", err)
+	}
+
+	// Create an SES instance
+	svc := ses.New(sess)
+
+	body, err := strconv.Unquote(string(req.Data))
```

Comment:
Why unquote? Is this a JSON String? If so, use JSON deserializer instead, so it can handle escaped quotes inside the string correctly.

Reply:
In SMTP and sendgrid binding it's used Unquoted method.

Must I change it?

Reply:
@artursouza  did you see my comment?

Reply:
@artursouza

Reply:
I checked and Unquote() will handle properly escaped quotes too: https://play.golang.org/p/LQHgvD9nZil